src/spatial_ref/srs.rs

```diff
@@ -7,6 +7,148 @@ use std::str::FromStr;
 
 use crate::errors::*;
 
+/// Options for [`CoordTransform::new_ex`].
+#[derive(Debug)]
+pub struct CoordTransformOptions {
+    inner: gdal_sys::OGRCoordinateTransformationOptionsH,
+}
+
+impl Drop for CoordTransformOptions {
+    fn drop(&mut self) {
+        unsafe { gdal_sys::OCTDestroyCoordinateTransformationOptions(self.inner) };
+        self.inner = ptr::null_mut();
+    }
+}
+
+impl CoordTransformOptions {
+    /// Creation options for [`CoordTransform`].
+    pub fn new() -> Result<CoordTransformOptions> {
+        let c_obj = unsafe { gdal_sys::OCTNewCoordinateTransformationOptions() };
+        if c_obj.is_null() {
+            return Err(_last_null_pointer_err(
+                "OCTNewCoordinateTransformationOptions",
+            ));
+        }
+        Ok(CoordTransformOptions { inner: c_obj })
+    }
+
+    /// Sets an area of interest.
+    ///
+    /// The west longitude is generally lower than the east longitude, except for areas of interest
+    /// that go across the anti-meridian.
+    ///
+    /// # Arguments
+    ///
+    /// - `west_longitude_deg` – West longitude (in degree). Must be in [-180,180]
+    /// - `south_latitude_deg` – South latitude (in degree). Must be in [-90,90]
+    /// - `east_longitude_deg` – East longitude (in degree). Must be in [-180,180]
+    /// - `north_latitude_deg` – North latitude (in degree). Must be in [-90,90]
+    pub fn set_area_of_interest(
+        &mut self,
+        west_longitude_deg: f64,
+        south_latitude_deg: f64,
+        east_longitude_deg: f64,
+        north_latitude_deg: f64,
+    ) -> Result<()> {
+        let ret_val = unsafe {
+            gdal_sys::OCTCoordinateTransformationOptionsSetAreaOfInterest(
+                self.inner,
+                west_longitude_deg,
+                south_latitude_deg,
+                east_longitude_deg,
+                north_latitude_deg,
+            )
+        };
+        if ret_val == 0 {
+            return Err(GdalError::OctError {
```

Comment:
I'm not sure how our errors are supposed to work, but this function returns an error message, e.g. `Invalid dfWestLongitudeDeg`. You can get it using `utils::_last_cpl_err`, but of course, that will return a `CplError` instead. So I don't really know what would be the best.

Reply:
I don't know why this OCT prefix is even used here (OGR Coordinate Transformation??), so it seems excessive to add another error type for it. `CplError` will do fine for now. Used this for the other setters as well, though these are (currently) infallible.